### PR TITLE
Standardize Logging Across EffortlessDB and EffortlessConfig

### LINKED ISSUE #2: Standardize logging
This issue is geared towards beginners and first-time contributors.

### Task Description:
The current logging for both [EffortlessConfig](https://github.com/bboonstra/Effortless/blob/main/effortless/configuration.py) and [EffortlessDB](https://github.com/bboonstra/Effortless/blob/main/effortless/effortless.py) is inconsistent. Some methods use `print`, others use `logger.log`, and others use `logger.debug`. Additionally, many outputs do not respect the `debug` configuration setting. 

We need to standardize the logging approach across these classes, ensuring that:
- All log outputs are managed through the same logging method.
- Logs are only displayed when the `debug` setting in the configuration is set to `True`.
- Missing or unclear debug messages should be added to functions where appropriate, providing meaningful context about the operation.

### Expected Behavior:
- The `EffortlessConfig` and `EffortlessDB` classes should use a single, consistent logging approach.
- All logging should occur **only** if `config.debug` is set to `True`.
- Functions missing useful debug messages should include additional logging that provides relevant and clear information for debugging.

### Notes:
- **Avoid copying external code.** We encourage contributors to explain their own approach. PRs that include in-depth explanations of the reasoning behind the solution will be prioritized for review.

### To Discourage AI Solutions:
When submitting a PR, please:
- Write a **brief explanation** of how your solution works, focusing on **why** you chose this approach.
- Discuss **potential challenges** you encountered and how you solved them.
- Add `Resolves #2` in the PR to link it to this issue.

If you do not follow these guidelines, your PR will be closed for AI.

### PR DESCRIPTION
**Pull request by Hemant-Shashikant-Yadav**
# Overview

This PR aims to standardize the logging approach across the `EffortlessDB` and `EffortlessConfig` classes. Currently, the logging is inconsistent, with some methods using `print`, others using `logger.log`, and others using `logger.debug`. Additionally, many log outputs do not respect the `debug` configuration setting, which can lead to unnecessary clutter in the output.

## Changes

1. **Consolidated Logging**: I've updated the code to use a single, consistent logging approach throughout the `EffortlessDB` and `EffortlessConfig` classes. All log outputs are now managed through the `logger` object.

2. **Respect `debug` Configuration**: The logging has been modified to only occur when the `config.debug` setting is set to `True`. This ensures that the log messages are only displayed when the user has explicitly enabled debug mode, improving the overall user experience.

## Rationale

The key reasons for these changes are:

1. **Consistency**: Having a consistent logging approach across the codebase makes it easier to understand and maintain the code, especially for new contributors.

2. **Configurability**: Respecting the `debug` configuration setting allows users to control the verbosity of the logging output, which is important for both development and production environments.

3. **Improved Debugging**: The addition of more meaningful debug messages provides better visibility into the inner workings of the `EffortlessDB` and `EffortlessConfig` classes, making it easier to diagnose and resolve issues.

## Potential Challenges

One potential challenge I encountered was ensuring that the logging approach was consistent across all the relevant methods and classes. It required careful review of the existing codebase to identify all the places where logging was used and ensure that they were updated to use the `logger` object.

Another challenge was determining the appropriate level of detail for the debug messages. I tried to strike a balance between providing enough information to aid in debugging without overwhelming the user with unnecessary details.

## Conclusion

This PR standardizes the logging approach in the `EffortlessDB` and `EffortlessConfig` classes, improving the overall maintainability and debuggability of the codebase. The changes ensure that logging is consistent, configurable, and provides meaningful information to help users troubleshoot issues.

Resolves #2

### CODE REVIEW

**Review by bboonstra — APPROVED** [review]
Thanks for taking the time to do some quality work!

File: effortless/effortless.py [hunk 1]
```diff
@@ -343,6 +362,7 @@ def add(self, entry: dict) -> None:
         data["content"].append(entry)
         self._write_db(data)
         self._handle_backup()
+        logger.debug(f"Entry added to database: {entry}")
```
Comment (bboonstra) [inline effortless/effortless.py:365]: This seems to generate significant performance overhead when large changes are made -- I would suggest truncating the message to a maximum length, which I am hoping will stop the unit tests from hanging. Once the unit tests pass, I'll merge these changes in!

**Review by bboonstra — CHANGES_REQUESTED** [review]
If you can improve the performance of your changes, they should be good to go.
image

File: effortless/effortless.py (hunk 1 above)
Comment (Hemant-Shashikant-Yadav) [reply effortless/effortless.py:365]: Yes I'll look into it.

**Comment by Hemant-Shashikant-Yadav** [comment]
Ok check and update soon

**Comment by bboonstra** [comment]
Any update? @Hemant-Shashikant-Yadav

**Pull request closed without merging**